### PR TITLE
Fix Duplicate Stearoyl-CoA Beta-Oxidation Reactions

### PR DESCRIPTION
**Pull request by Devlin-Moyer**
#### Main improvements in this PR:

As proposed in #802:
- Removes `MAR00776` for being a duplicate of `MAR03115` + `MAR03116` + `MAR03117` + `MAR03118`
- Removes `MAR00778` for being a duplicate of  `MAR03116` + `MAR03117` + `MAR03118` + `MAR03121` + `MAR03122` + `MAR03123` + `MAR03125` + `MAR03128` + `MAR03129` + `MAR03130` + `MAR03132` +`MAR03135` + `MAR03136` + `MAR03137` + `MAR03139` + `MAR03142` + `MAR03143` + `MAR03144` + `MAR03146`

I noticed that `MAR00776` had references associated with it but `MAR03115`, `MAR03116`, `MAR03117`, and `MAR03118` didn't, but the two references associated with `MAR00776` were about ACADS and ACADM, neither of which can actually catalyze the first round of beta-oxidation of stearoyl-CoA (at 18 carbons, it's too long), so I instead associated those four reactions with `PMID:20490924`, a review of mitochondrial beta-oxidation. `MAR03121` onwards all had several references already associated with them, so I left those alone.

**I hereby confirm that I have:**
- [X] Tested my code on my own computer for running the model
- [X] Selected `develop` as a target branch
- [X] Any removed reactions and metabolites have been moved to the corresponding deprecated identifier lists

### CODE REVIEW

**Comment by Devlin-Moyer** [comment]
it would be easier to figure out what went wrong with the metabolic tasks if the test indicated which tasks failed rather than just saying that the test as a whole failed (if this information is in the output somewhere and I'm just not seeing it, let me know)

**Comment by feiranl** [comment]
@JHL-452b  Could you help check this?

**Comment by JHL-452b** [comment]
> it would be easier to figure out what went wrong with the metabolic tasks if the test indicated which tasks failed rather than just saying that the test as a whole failed (if this information is in the output somewhere and I'm just not seeing it, let me know)

Yeah. In the [current code](https://github.com/SysBioChalmers/Human-GEM/blob/develop/code/test/testMetabolicTasks.m), it would exit the matlab program directly if the task failed without any feedback. This suggestion should be incorporated into plans for subsequent improvements. 

However, in this bug, I tested on my own computer and finded that the task of `Growth on Ham's media (biomass production)` was failed. The cause of the failure is being analyzed, and will provide feedback as soon as possible when there are conclusions.

**Comment by JHL-452b** [comment]
Actually, the failed task seems to have nothing to do with the modification of this PR due to same problem occured when  I re-run [the task](https://github.com/SysBioChalmers/Human-GEM/actions/runs/8981622261) for `./develop/Human-GEM`.

I tried changing the solver to `gurobi`, and all tasks succeeded as you can see below:

```matlab
>> taskType = 'essential';
>> setRavenSolver('glpk')
>> status = testMetabolicTasks(taskType)
Boundary metabolites were added to 1660 reactions.
New (boundary) versions of 1660 metabolites were added to the model.

FAIL: [GR] Growth on Ham's media (biomass production)
status =
     0

>> setRavenSolver('gurobi')
>> status = testMetabolicTasks(taskType)
Boundary metabolites were added to 1660 reactions.
New (boundary) versions of 1660 metabolites were added to the model.

Suceeded with essential tasks.
status =
     1
```

I don't know why RAVEN cannot automatically change the solver to `gurobi` in the running code of those tasks now. If we don't resolve it, all subsequent PR will fail. @feiranl

**Comment by edkerk** [comment]
See https://github.com/SysBioChalmers/Human-GEM/issues/821#issuecomment-2107505146 for consideration regarding gurobi/glpk.

File: model/reactions.tsv [hunk 1]
```diff
@@ -2624,26 +2624,26 @@ rxns	rxnKEGGID	rxnBiGGID	rxnEHMNID	rxnHepatoNET1ID	rxnREACTOMEID	rxnRecon3DID	rx
...
-"MAR03142"	"R04754"	"r1451;FAOXC10C10OHm;FAOXC10080m"	""	""	"R-HSA-77345"	"HMR_3142;r1451;FAOXC10C10OHm;FAOXC10C8m"	"MNXR109095"	"HMR_3142"	"RCR12645"	""	0	""	"RHEA:28354"	"HMR_3142;r1451;MAR02373;FAOXC10C10OHm;MAR05019;FAOXC10C8m;MAR05024"
-"MAR03143"	"R04744"	"FAOXC10C10OHm;FAOXC101C8m;FAOXC10080m"	"R04744X"	"r0728"	""	"r0728;HMR_3422;FAOXC10C10OHm;FAOXC101C8m;FAOXC10C8m"	"MNXR97889"	"HMR_3143;HMR_3422"	"RCR12646;RCR12667"	""	0	""	"RHEA:31191"	"HMR_3143;HMR_3422;MAR03422;FAOXC10C10OHm;MAR05019;FAOXC101C8m;MAR04967;FAOXC10C8m;MAR05024"
-"MAR03144"	"R04743"	"FAOXC101C8m;FAOXC10080m"	"R04743X"	"r0726"	""	"r0726;HMR_3422;FAOXC101C8m;FAOXC10C8m"	"MNXR100545"	"HMR_3144;HMR_3422"	"RCR12647;RCR12667"	""	0	""	"RHEA:31187"	"HMR_3144;HMR_3422;MAR03422;FAOXC101C8m;MAR04967;FAOXC10C8m;MAR05024"
-"MAR03146"	"R03778"	"FAOXC101C8m;FAOXC10080m"	"R03778X"	"r0634"	""	"r0634;HMR_3422;FAOXC101C8m;FAOXC10C8m"	"MNXR95198"	"HMR_3146;HMR_3422"	"RCR12648;RCR12667"	""	0	"RHEA:31089"	"RHEA:31087"	"HMR_3146;HMR_3422;MAR03422;FAOXC101C8m;MAR04967;FAOXC10C8m;MAR05024"
+"MAR03115"	""	"FAOXC180"	""	""	""	"HMR_3115;FAOXC180"	""	"HMR_3115"	"RCR12629"	""	0	""	""	"HMR_3115;FAOXC180;MAR00776"
```
[the remark this answers is not in the document]
Comment (Devlin-Moyer) [reply model/reactions.tsv:2627]: `MAR00776` was associated with `MNXR99313`: 1 hexadecanoyl-CoA + 3 H(+) + 1 NADH + 1 acetyl-CoA + 1 FADH2 <-> 1 octadecanoyl-CoA + 1 FAD + 1 CoA + 1 NAD(+) + 1 H2O, so I shouldn't have associated it with `MAR03117` or `MAR03118`. I appear to have not had that issue with any of the other reactions

[the remark this answers is not in the document]
Comment (JHL-452b) [reply model/reactions.tsv:2627]: OK, it looks great to me. (*^_^*)

**Comment by feiranl** [comment]
@Devlin-Moyer @edkerk  Is this PR ready to be merged?

**Comment by edkerk** [comment]
Yes, but I'd like to merge-in `develop` when #883 is approved.

**Pull request merged**